src/fsharp/PrettyNaming.fs

```diff
@@ -318,38 +482,29 @@ module internal Microsoft.FSharp.Compiler.PrettyNaming
 
     let private chopStringTo (s:string) (c:char) =
         (* chopStringTo "abcdef" 'c' --> "def" *)
-        if s.IndexOf c <> -1 then
-            let i =  s.IndexOf c + 1
+        match s.IndexOf c with
+        | -1 -> s
+        | idx ->
+            let i = idx + 1
             s.Substring(i, s.Length - i)
-        else
-            s
 
     /// Try to chop "get_" or "set_" from a string
     let TryChopPropertyName (s: string) =
-        // extract the logical name from any mangled name produced by MakeMemberDataAndMangledNameForMemberVal 
-        let s = 
-            if s.StartsWith("get_", System.StringComparison.Ordinal) || 
-               s.StartsWith("set_", System.StringComparison.Ordinal) 
-            then s 
-            else chopStringTo s '.'
-
-        if s.Length <= 4 || (let s = s.Substring(0,4) in s <> "get_" && s <> "set_") then
-            None
-        else 
-            Some(s.Substring(4,s.Length - 4) )
-
-
+        // extract the logical name from any mangled name produced by MakeMemberDataAndMangledNameForMemberVal
+        if s.Length <= 4 then None else
+        let s = chopStringTo s '.'
```

Comment:
Apologies, my earlier suggestion to use this simplification was bogus.  This now fails/causes crash for identifiers with `.` in them (e.g. backtick identifiers). I'll fix before merging...